msteams-platform/messaging-extensions/how-to/authentication/msgex-sso-graph-api1.md

```diff
@@ -0,0 +1,144 @@
+
+    The options for Graph permissions display.
+
+5. Select **Delegated permissions** to view the list of permissions.
+
+   :::image type="content" source="../../../assets/images/authentication/teams-sso-tabs/delegated-permission.png" alt-text="Delegated permissions." border="true":::
+
+6. Select relevant permissions for your app, and then select **Add permissions**.
+
+   :::image type="content" source="../../../assets/images/authentication/teams-sso-tabs/select-permission.png" alt-text="Select permissions." border="true":::
+
+    You can also enter the permission name in the search box to find it.
+
+    A message pops up on the browser stating that the permissions were updated.
+
+   :::image type="content" source="../../../assets/images/authentication/teams-sso-tabs/updated-permission-msg.png" alt-text="Permissions updated message." border="false":::
+
+    The added permissions are displayed in the **API permissions** page.
+
+   :::image type="content" source="../../../assets/images/authentication/teams-sso-tabs/configured-permissions.png" alt-text="API permissions are configured." border="true":::
+
+    You've configured your app with Microsoft Graph permissions.
+
+## Configure authentication for different platforms
+
+Configuration for authentication depends on the platform or device where you want to target your app. You may need to configure redirect URIs, specific authentication settings, or details specific to the platform.
+
+> [!NOTE]
+>
+> - If your bot app hasn't been granted IT admin consent, app users have to provide consent the first time they use your app on a different platform.
+> - Implicit grant is not required if SSO is enabled on a bot app.
+
+You can configure authentication for multiple platforms as long as the URL is unique.
+
+### To configure authentication for a platform
+
+1. Open the app you registered in the [Azure portal](https://ms.portal.azure.com/).
+
+1. Select **Manage** > **Authentication** from the left pane.
+
+    :::image type="content" source="../../../assets/images/authentication/teams-sso-tabs/azure-portal-platform.png" alt-text="Authenticate for platforms" border="true":::
+
+    The **Platform configurations** page appears.
+
+1. Select **+ Add a platform**.
+
+    :::image type="content" source="../../../assets/images/authentication/teams-sso-tabs/add-platform.png" alt-text="Add a platforms" border="true":::
+
+    The **Configure platforms** page appears.
+
+1. Select the platform that you want to configure for your bot app. You can choose the platform type from web or SPA.
+
+    :::image type="content" source="../../../assets/images/authentication/teams-sso-tabs/configure-platform.png" alt-text="Select web platform" border="true":::
+
+    You can configure multiple platforms for a particular platform type. Ensure that the redirect URI is unique for every platform you configure.
```

Comment:
Where is this uniqueness constraint coming from? 

Also can I configure multiple platforms for a particular _platform_ or an _application_?